src/rprocessing/RLangPApplet.java

```diff
@@ -342,18 +342,31 @@ public void focusLost() {
     super.focusLost();
     this.renjinEngine.put("focused",super.focused);
   }
+
+  @Override
+  public void mouseMoved() {
+    wrapMouseVariables();
+  }
+
+  private void wrapMouseVariables() {
+    this.renjinEngine.put("mouseX", mouseX);
+    this.renjinEngine.put("mouseY", mouseY);
+    this.renjinEngine.put("pmouseX", pmouseX);
+    this.renjinEngine.put("pmouseY", pmouseY);
+    //this.renjinEngine.put("mouseButton", mouseButton);
+  }
   
   /**
    * Set Environment variables in R top context.
    */
   protected void wrapProcessingVariables() {
     log("Wrap Processing built-in variables into R top context.");
     // TODO: Find some ways to push constants into R.
+    wrapMouseVariables();
     this.renjinEngine.put("width", width);
     this.renjinEngine.put("height", height);
     this.renjinEngine.put("displayWidth", displayWidth);
     this.renjinEngine.put("displayHeight", displayHeight);
-    // this.renjinEngine.put("focused", focused);
```

Comment:
Yes, I enabled that line in 912d23b -- before, `processing$focused` worked, but `focused` did not. Now `focused` works. We should be able to merge the mouseX / mouseY code and leave the line -- unless I'm misunderstanding something.